docs/plugins/exclude_priority_table.py

```diff
@@ -0,0 +1,63 @@
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Union
+
+from pydantic import BaseModel, Field
+
+
+@dataclass
+class ExcludeSetting:
+    name: str
+    value: Union[bool, set]
+
+    @property
+    def markdown_str(self) -> str:
+        return f'{self.name}={self.value}'
+
+    @property
+    def kwargs_dict(self) -> Dict[str, Union[str, bool, set]]:
+        return {} if self.name == 'not_specified' else {self.name: self.value}
+
+
+model_dump_exclude_variants_settings: List[ExcludeSetting] = [
+    ExcludeSetting(name='exclude', value={'name'}),
+    ExcludeSetting(name='include', value={}),
+    ExcludeSetting(name='exclude_none', value=True),
+    ExcludeSetting(name='exclude_defaults', value=True),
+    ExcludeSetting(name='exclude_unset', value=True),
+]
+
+
+base_markdown = """
+```py
+from typing import Optional
+
+from pydantic import BaseModel, Field
+
+class Dog(BaseModel):
+    name: Optional[str] = Field(default='Unspecified', exclude=False)
+
+"""
+
+
+class Dog(BaseModel):
+    name: Optional[str] = Field(default='Unspecified', exclude=False)
+
+
+def _build_exclude_example_markdown(exclude_setting: ExcludeSetting) -> str:
+    modified_markdown = base_markdown
+    for init_kws in [{'name': 'Ralph'}, {'name': 'Unspecified'}, {'name': None}, {}]:
+        my_dog = Dog(**init_kws)
+
+        modified_markdown += f'my_dog = Dog(**{init_kws})\n'
+        modified_markdown += f'print(my_dog.model_dump({exclude_setting.markdown_str}))\n'
+        modified_markdown += f'#> {my_dog.model_dump(**exclude_setting.kwargs_dict)}\n'
+
+    modified_markdown += '```'
+
+    return modified_markdown
```

Comment:
Do we need this to be automated? 🤔 

All of these look kind a the same to me... 🤔 

<img width="679" alt="Screenshot 2023-09-14 at 11 06 34" src="https://github.com/pydantic/pydantic/assets/7353520/b5651304-7a9e-44b4-b344-2c2d705d9ed1">

Reply:
Each example is pretty similar for the `exclude` and `include` tabs, but they show that `exclude` at the `model_dump` level takes priority over that at the field level. The `exclude_***` tabs have more variety in their output though.